// oura-api-v2-client/dist/cjs/domains/services/DailyActivityService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DailyActivityService = void 0;
const OpenAPI_1 = require("../core/OpenAPI");
const request_1 = require("../core/request");
class DailyActivityService {
    /**
     * Get Daily Activity
     * Returns Oura Daily Activity data for the specified Oura user within a given timeframe
     * @param startDate
     * @param endDate
     * @param nextToken
     * @returns DailyActivityResponse OK
     * @throws ApiError
     */
    static dailyActivityRouteDailyActivityGet(startDate, endDate, nextToken) {
        return request_1.request(OpenAPI_1.OpenAPI, {
            method: 'GET',
            url: '/v2/usercollection/daily_activity',
            query: {
                'start_date': startDate,
                'end_date': endDate,
                'next_token': nextToken,
            },
            errors: {
                400: `Query Parameter Validation Error`,
                426: `Minimum App Version Error`,
            },
        });
    }
}
exports.DailyActivityService = DailyActivityService;
