// oura-api-v2-client/dist/cjs/domains/services/HeartRateService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HeartRateService = void 0;
const OpenAPI_1 = require("../core/OpenAPI");
const request_1 = require("../core/request");
class HeartRateService {
    /**
     * Get Heart Rate
     * Returns available heart rate data for a specified Oura user within a given timeframe
     * @param startDatetime
     * @param endDatetime
     * @param nextToken
     * @returns HeartRateResponse OK
     * @throws ApiError
     */
    static heartrateRouteHeartrateGet(startDatetime, endDatetime, nextToken) {
        return request_1.request(OpenAPI_1.OpenAPI, {
            method: 'GET',
            url: '/v2/usercollection/heartrate',
            query: {
                'start_datetime': startDatetime,
                'end_datetime': endDatetime,
                'next_token': nextToken,
            },
            errors: {
                400: `Query Parameter Validation Error`,
                426: `Minimum App Version Error`,
            },
        });
    }
}
exports.HeartRateService = HeartRateService;
