// oura-api-v2-client/dist/cjs/domains/services/SessionsService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionsService = void 0;
const OpenAPI_1 = require("../core/OpenAPI");
const request_1 = require("../core/request");
class SessionsService {
    /**
     * Get Sessions
     * Returns available Oura session data for the specified Oura user within a given timeframe
     * @param startDate
     * @param endDate
     * @param nextToken
     * @returns SessionResponse OK
     * @throws ApiError
     */
    static sessionsRouteSessionGet(startDate, endDate, nextToken) {
        return request_1.request(OpenAPI_1.OpenAPI, {
            method: 'GET',
            url: '/v2/usercollection/session',
            query: {
                'start_date': startDate,
                'end_date': endDate,
                'next_token': nextToken,
            },
            errors: {
                400: `Query Parameter Validation Error`,
                426: `Minimum App Version Error`,
            },
        });
    }
}
exports.SessionsService = SessionsService;
