// oura-api-v2-client/dist/cjs/domains/services/WorkoutsService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkoutsService = void 0;
const OpenAPI_1 = require("../core/OpenAPI");
const request_1 = require("../core/request");
class WorkoutsService {
    /**
     * Get Workouts
     * Returns available Oura workout data for the specified Oura user within a given timeframe
     * @param startDate
     * @param endDate
     * @param nextToken
     * @returns WorkoutResponse OK
     * @throws ApiError
     */
    static workoutsRouteWorkoutGet(startDate, endDate, nextToken) {
        return request_1.request(OpenAPI_1.OpenAPI, {
            method: 'GET',
            url: '/v2/usercollection/workout',
            query: {
                'start_date': startDate,
                'end_date': endDate,
                'next_token': nextToken,
            },
            errors: {
                400: `Query Parameter Validation Error`,
                426: `Minimum App Version Error`,
            },
        });
    }
}
exports.WorkoutsService = WorkoutsService;
