// oura-api-v2-client/dist/esm/src/index.js
import ApiClient from "./apiClient";
import { URLSearchParams } from "url";
export default class OuraApiV2Client {
    constructor(accessToken, api) {
        this.api = api ?? new ApiClient(accessToken);
    }
    async dailyActivity(props) {
        const parameters = new URLSearchParams();
        if (props?.startDate) {
            parameters.set("start_date", props.startDate);
        }
        if (props?.endDate) {
            parameters.set("end_date", props.endDate);
        }
        if (props?.nextToken) {
            parameters.set("next_token", props.nextToken);
        }
        const response = await this.api.fetch("/v2/usercollection/daily_activity", parameters.toString());
        return response.body;
    }
    async personalInfo() {
        const response = await this.api.fetch("/v2/usercollection/personal_info");
        return response.body;
    }
    async heartRate(props) {
        const parameters = new URLSearchParams();
        if (props?.startDate_timestamp) {
            parameters.set("start_date", props.startDate_timestamp);
        }
        if (props?.endDate_timestamp) {
            parameters.set("end_date", props.endDate_timestamp);
        }
        if (props?.nextToken) {
            parameters.set("next_token", props.nextToken);
        }
        const response = await this.api.fetch("/v2/usercollection/heartrate", parameters.toString());
        return response.body;
    }
    async sessions(props) {
        const parameters = new URLSearchParams();
        if (props?.startDate) {
            parameters.set("start_date", props.startDate);
        }
        if (props?.endDate) {
            parameters.set("end_date", props.endDate);
        }
        if (props?.nextToken) {
            parameters.set("next_token", props.nextToken);
        }
        const response = await this.api.fetch("/v2/usercollection/session", parameters.toString());
        return response.body;
    }
    async tags(props) {
        const parameters = new URLSearchParams();
        if (props?.startDate) {
            parameters.set("start_date", props.startDate);
        }
        if (props?.endDate) {
            parameters.set("end_date", props.endDate);
        }
        if (props?.nextToken) {
            parameters.set("next_token", props.nextToken);
        }
        const response = await this.api.fetch("/v2/usercollection/tag", parameters.toString());
        return response.body;
    }
    async workouts(props) {
        const parameters = new URLSearchParams();
        if (props?.startDate) {
            parameters.set("start_date", props.startDate);
        }
        if (props?.endDate) {
            parameters.set("end_date", props.endDate);
        }
        if (props?.nextToken) {
            parameters.set("next_token", props.nextToken);
        }
        const response = await this.api.fetch("/v2/usercollection/workout", parameters.toString());
        return response.body;
    }
    /**
     * @deprecated
     * @see https://cloud.ouraring.com/docs/
     */
    async sleep_v1(props) {
        const parameters = new URLSearchParams();
        if (props?.startDate) {
            parameters.set("start", props.startDate);
        }
        if (props?.endDate) {
            parameters.set("end", props.endDate);
        }
        const response = await this.api.fetch("/v1/sleep", parameters.toString());
        return response.body;
    }
    /**
     * @deprecated
     * @see https://cloud.ouraring.com/docs/
     */
    async readiness_v1(props) {
        const parameters = new URLSearchParams();
        if (props?.startDate) {
            parameters.set("start", props.startDate);
        }
        if (props?.endDate) {
            parameters.set("end", props.endDate);
        }
        const response = await this.api.fetch("/v1/readiness", parameters.toString());
        return response.body;
    }
}
